### DISCUSSION TITLE
RTR_DeleteSession 400 Error

### DISCUSSION THREAD

**Discussion by cactusjuan** [CrowdStrike/falconpy#712] [Q&A] [answered]
I have made sure that I have sessions using the RTR_ListAllSessions but when I go to delete any of them, using either the service or uber class RTR_DeleteSession, I keeping getting `'errors': [{'code': 400, 'message': 'could not find session'}]`. Wanted to verify if there are any current bugs with this command?

**Comment by jshcodes** [comment]
We can check quickly by having you try executing a command using something like the [bulk_execute](https://github.com/CrowdStrike/falconpy/blob/main/samples/rtr/bulk_execute.py) sample. Are you able to execute a `ls -al` command against a target using this script?

```shell
python3 bulk_execute.py -k CLIENT_ID_HERE -s CLIENT_SECRET_HERE -f TARGET_HOSTNAME_HERE -c "ls -al"
```

Here's the output I get when I try it:
```shell
python3 bulk_execute.py -k CLIENT_ID_HERE -s CLIENT_SECRET_HERE -f TARGET_HOSTNAME -c "ls -al"
Starting sessions with target hosts.
Session with falconpy-unit-testing started successfully. [REDACTED-6960-47dd-b557-REDACTED]

Executing command (`ls -al`) against target hosts.

Closing sessions with target hosts.
Session REDACTED-6960-47dd-b557-REDACTED deleted successfully.

ENDPOINT-HOSTNAME
total 16
dr-xr-xr-x  18 root root  257 Nov 18  2021 .
dr-xr-xr-x  18 root root  257 Nov 18  2021 ..
-rw-r--r--   1 root root    0 Jul 14  2021 .autorelabel
lrwxrwxrwx   1 root root    7 Jul  6  2021 bin -> usr/bin
dr-xr-xr-x   4 root root 4096 Jul  6  2021 boot
drwxr-xr-x  15 root root 2820 May 13 04:40 dev
drwxr-xr-x  85 root root 8192 Sep 16  2021 etc
drwxr-xr-x   3 root root   22 Jul 14  2021 home
lrwxrwxrwx   1 root root    7 Jul  6  2021 lib -> usr/lib
lrwxrwxrwx   1 root root    9 Jul  6  2021 lib64 -> usr/lib64
drwxr-xr-x   2 root root    6 Jul  6  2021 local
drwxr-xr-x   2 root root    6 Apr  9  2019 media
drwxr-xr-x   2 root root    6 Apr  9  2019 mnt
drwxr-xr-x   6 root root   64 Sep 16  2021 opt
dr-xr-xr-x 119 root root    0 May 13 04:40 proc
dr-xr-x---   5 root root  169 Jun  7 18:09 root
drwxr-xr-x  30 root root 1120 Jul  1 20:05 run
lrwxrwxrwx   1 root root    8 Jul  6  2021 sbin -> usr/sbin
drwxr-xr-x   2 root root    6 Apr  9  2019 srv
dr-xr-xr-x  13 root root    0 Jun  8 20:23 sys
drwxrwxrwt   8 root root  172 Jul  1 20:05 tmp
drwxr-xr-x  13 root root  155 Jul  6  2021 usr
drwxr-xr-x  19 root root  269 Jul 14  2021 var
```

**Comment by cactusjuan** [comment]
I was able to run that script successfully. I was using [RTR_InitSession](https://falconpy.io/Service-Collections/Real-Time-Response.html#rtr_initsession). Is this not the correct way to start a session with a single host?

**Comment by jshcodes** [comment, marked as answer]
You can use either operation, [RTR_InitSession](https://falconpy.io/Service-Collections/Real-Time-Response.html#rtr_initsession) or [BatchInitSession](https://www.falconpy.io/Service-Collections/Real-Time-Response.html#batchinitsessions) to create a new session with a singular host.

We have an example of deleting a session created with RTR_InitSession [here](https://github.com/CrowdStrike/falconpy/blob/main/samples/rtr/pony/my_little_rtr.py#L195).

If you sanitize any API keys and AID / CID values from your code, and reply back with the section of code that's generating the error, we can check it out and see if anything looks out of place.
